dist/lucca-ui.js

```diff
@@ -872,6 +873,437 @@
+			var deferred = $q.defer();
+
+			reinitTimeout();
+
+			// launch new timeout 
+			timeout.count = $timeout(function() {
+				getCountAsync(input).then(
+					function (count) {
+						deferred.resolve(count);
+					},
+					function (error) {
+						// HANDLE ERROR
+						deferred.reject(error);
+					}
+				);
+			}, delay);
+			return deferred.promise;
+		};
+
+		var getCountAsync = function(input) {
+			var deferred = $q.defer();
+			var dtContractEnd = "&dtcontractend=since," + moment().format("YYYY-MM-DD") + ",null";
+			var query = "/api/v3/users?name=like," + input + "&fields=collection.count" + ($scope.showFormerEmployees ? "" : dtContractEnd); // query for count
+
+			delete timeout.count;
+			$http.get(query).then(
+				function(response) {
+					deferred.resolve(response.data.data.count);
+				},
+				function(error) {
+					deferred.reject(error);
+				}
+			);
+			return deferred.promise;
+		};
+
+		var addOverflowMessage = function(users, maxNbUsers) {
+			var firstUsers = _.first(users, MAX_COUNT);
+			firstUsers.push({ overflow: MAX_COUNT + "/" + maxNbUsers });
+			return firstUsers;
+		};
+
+		var updateOverflowMessage = function(users, nbUsers, maxNbUsers) {
+			_.last(users).overflow = nbUsers + "/" + maxNbUsers;
+			return users;
+		};
+
+		/********************/
+		/***** HOMONYMS *****/
+		/********************/
+
+		var hasHomonyms = function(users) {
+			var usersWithoutHomonyms = _.uniq(users, function(user) { return (user.firstName + user.lastName); });
+
+			if (usersWithoutHomonyms.length < users.length) {
+				return true;
+			}
+			return false;
+		};
+
+		var handleHomonyms = function(users) {
+			var homonyms = _.where(users, { hasHomonyms: true });
+			var homonymsArray = [];
+
+			getHomonymsPropertiesAsync(homonyms).then(
+				function (response) {
+					// Add homonyms properties for each user
+					_.each(homonyms, function(user, item) {
+						// Get the returned user
+						var userWithProps = response[item].data.data.items[0];
+
+						// Add each property to the user
+						_.each($scope.properties, function (prop) {
+							var newProp = prop.split('.')[0];
+							user[newProp] = userWithProps[newProp];
+						});
+					});
+				},
+				function(error) {
+					// HANDLE ERROR
+				}
+			);
+		};
+
+		var tagHomonyms = function(users) {
+			_.each(users, function (user, index) {
+				var rest = _.rest(users, index + 1);
+				_.each(rest, function (otherUser) {
+					if ((user.firstName === otherUser.lastName) && (user.lastName === otherUser.lastName)) {
+						user.hasHomonyms = true;
+						otherUser.hasHomonyms = true;
+					}
+				});
+
+				// Search homonyms in selected users (used by UserPickerMultiple only)
+				_.each($scope.selected.users, function (selectedUser) {
+					if ((user.firstName === selectedUser.firstName) && (user.lastName === selectedUser.lastName)) {
+						user.hasHomonyms = true;
+						selectedUser.hasHomonyms = true;
+					}
+				});
```

Comment:
I don't like the fact that the homonyms module has to have specific logic for multi-select shenanigans.

if after removing the users corresponding to `$scope.selected.users` from `filteredUsers` (line ~1k) you then add `$scope.selected.users` to `filteredUsers` it might work. the ui-select nultiple wont display a choice corresponding to something selected so they won't appear in the choices and it allows you to consider `filteredUsers` as being the list of available choices for the rest of the treatment